### PR TITLE
Fix part of #210: Added file 'How to find a commit which introduced a bug'

### PR DESCRIPTION
**Pull request by satvshr**
Issue #210

### CODE REVIEW

File: Debugging.md
Comment (seanlip) [inline Debugging.md]: Debugging --> Finding

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Using --> using

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Drop "to"

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Use backticks for code: `

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Use markdown constructs rather than html (strong/span/underline etc.) Ditto throughout the entire file.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: ...on that commit, then we know that the problem happened between that commit and the current one. Go ahead with one of the approaches below to find the commit which introduced the error.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: I think points 3 and 4 should be formatted and rewritten as a single step with two alternatives.

Also, for this one, let's say: ... on that commit, then we know that the problem happened before the earlier commit. In that case, report that ...

Also change "the commit" to "the earliest commit you checked".

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Please drop all html constructs from this file.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: are --> were

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md:28]: Can we put this one first? It's going to be the more common approach (before git blame on cmdline).

Also, maybe explain: This is useful if you know which specific line of code caused the error.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Navigate to the file. (In this example, we need to ...)


Btw, do you actually need to bold every single line? It'll make it harder to read -- let's instead reserve bolding for specific parts of the text we want to emphasize.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Maybe also explain: This is more useful if you don't know which line of code is causing the error (otherwise, we recommend using the "git blame" approach above instead).

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: steps --> step

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @satvshr! Took a pass and left comments, PTAL.

**Comment by seanlip** [comment]
@satvshr Please follow our [PR instructions](https://github.com/oppia/oppia/wiki/Make-a-pull-request). Reply to all comments, and do not resolve them (see step 5 in the previous link and please read it carefully). Please do this before requesting a follow-up review.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done (removed code tag but didnt add a backtick as it wasnt needed)

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md:28]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

**Comment by satvshr** [comment]
> @satvshr Please follow our [PR instructions](https://github.com/oppia/oppia/wiki/Make-a-pull-request). Reply to all comments, and do not resolve them (see step 5 in the previous link and please read it carefully). Please do this before requesting a follow-up review.

@seanlip  Unresolved all the comments made by you and already made those changes a while back, sorry!

**Comment by satvshr** [comment]
> > @satvshr Please follow our [PR instructions](https://github.com/oppia/oppia/wiki/Make-a-pull-request). Reply to all comments, and do not resolve them (see step 5 in the previous link and please read it carefully). Please do this before requesting a follow-up review.
> 
> @seanlip Unresolved all the comments made by you and already made those changes a while back, sorry!

Hey @seanlip, could you please recheck all my changes and let me know if any more modifications need to be made? Exams start soon so I just want to be done with 1 pr by then, thanks!

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Why do you still have html constructs (&lt;) here and below? 

Perhaps just put them in parentheses: (...)

File: Debugging.md
Comment (seanlip) [inline Debugging.md]: Let's make this the second link in the list, since it's a more general doc.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Why do you need to bold everything? As mentioned before try to use bold only for important keywords, otherwise it's hard to read.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: Also for this link, use the short hash similar to below -- easier to read. (Try and preview your changes using the inbuilt GitHub functionality -- these issues will become obvious.)

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: You will need to indent this more so it fits under point 1. Again please preview your work.

Ditto for pretty much everything below.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (seanlip) [inline How-to-find-the-commit-which-introduced-a-bug.md]: ...navigate to the...

Also why is the numbering here different from above? Let's be consistent.

**Review by seanlip — CHANGES_REQUESTED** [review]
@satvshr Thanks for the changes. I feel like this PR lacks a bit of quality control. Could you please preview your changes using the github markdown editor preview functionality, and make sure everything looks good, the indentation is correct, the numbering is consistent between both sections, etc.?

In your next request for review, please show screenshots of the preview -- it is important to always test any changes from the end-user perspective and apply good judgment before asking others to look at them. Thanks.

File: Debugging.md
Comment (satvshr) [reply Debugging.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Removed the constructs and added paranthesis

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done. Same image as [above](https://github.com/oppia/oppia-web-developer-docs/pull/219#discussion_r1441466277)

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: "I used Preview to write the article but decided to retain most of the content as it appeared in the google doc, assuming it was the preferred format.

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: Done

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: image

Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: [Improved formatting](https://github.com/oppia/oppia-web-developer-docs/pull/219#discussion_r1441466277)(same image as one used above)

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: image

File: How-to-find-the-commit-which-introduced-a-bug.md
Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md:155]: image

Comment (satvshr) [reply How-to-find-the-commit-which-introduced-a-bug.md]: image

**Comment by satvshr** [comment]
Resolved all issues again, sorry for making so many mistakes.
Should I make a pr @seanlip ?

**Comment by seanlip** [comment]
You should update the current PR. Your changes are not showing in the "Files Changed" tab: https://github.com/oppia/oppia-web-developer-docs/pull/219/files

Once you have updated the PR you can re-request a review. (I don't know what you mean by "Should I make a pr" -- this is already a PR.)

See [our guide to making PRs](https://github.com/oppia/oppia/wiki/Make-a-pull-request) (particularly step 5) for info about this process and please try to follow it in all future PRs. Thanks.

**Comment by satvshr** [comment]
> Once you have updated the PR you can re-request a review. (I don't know what you mean by "Should I make a pr" -- this is already a PR.)

You told me to wait for your approval to make one again, that's why I was asking you about it.Nevertheless, could you please review the changes again @seanlip?

**Comment by seanlip** [comment]
@satvshr Looks generally fine, the only issue I see is that, under "Using GitHub website", the points below item 4 should be indented so that they're under item 4. Otherwise I think that's it!

> You told me to wait for your approval to make one again, that's why I was asking you about it.Nevertheless, could you please review the changes again @seanlip?

Where did I tell you that? Normally, devs don't create new PRs every time a review is needed, they update the existing PR. See our contribution guidelines in [step 5 here](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve) for reference.

**Comment by seanlip** [comment]
@satvshr If possible, could you reply to the previous comment and make the changes soonish? I would like to refer to this page in a PR review I'm doing -- thanks!

**Comment by satvshr** [comment]
> @satvshr If possible, could you reply to the previous comment and make the changes soonish? I would like to refer to this page in a PR review I'm doing -- thanks!

Hey Sean, Sorry I didnt see your message, exams started and have been busy with that. Give me a sec

**Comment by satvshr** [comment]
> > @satvshr If possible, could you reply to the previous comment and make the changes soonish? I would like to refer to this page in a PR review I'm doing -- thanks!
> 
> Hey Sean, Sorry I didnt see your message, exams started and have been busy with that. Give me a sec

Edit: @seanlip I guess I am done? Kindly have a look!

**Review by seanlip — APPROVED** [review]
Yup, looks good, thanks @satvshr!

**Comment by seanlip** [comment]
Merging this now, could you please see the last part of  https://github.com/oppia/oppia-web-developer-docs/issues/210 as well? Thanks!

**Comment by seanlip** [comment]
Also congrats on your very first PR to Oppia! :tada:

**Comment by satvshr** [comment]
> Also congrats on your very first PR to Oppia! 🎉

Thank you! Hopefully next time I wont be needing an extremely long thread with a ton of misunderstandings from my side to make one :p

**Pull request merged**
